packages/react-dom/src/__tests__/ReactDOMFizzStaticBrowser-test.js

```diff
@@ -1316,4 +1550,146 @@ describe('ReactDOMFizzStaticBrowser', () => {
         '</head><body><div>Hello</div></body></html>',
     );
   });
+
+  // @gate enablePostpone
+  it(
+    'during a resume, loading prerendered data from cache should not block ' +
+      'sibling Suspense trees from rendering in parallel',
+    async () => {
+      let prerendering = true;
+
+      function DynamicText({text}) {
+        if (prerendering) {
+          React.unstable_postpone();
+        }
+        Scheduler.log(text);
+        return text;
+      }
+
+      function readTextAndLog(text) {
+        readText(text);
+        Scheduler.log(text);
+        return text;
+      }
+
+      function Toolbar() {
+        // This read is in the parent path of Toolbar content, so during the
+        // resume it will run again. During the resume, it should be cached, but
+        // reading from the cache may be async so it could still suspend,
+        // creating a waterfall. Because it's cached, it's no big deal, but
+        // regardless it shouldn't block unrelated siblings (MainContent)
+        // from rendering.
+        const toolbarText = readTextAndLog('Toolbar shell');
+
+        return (
+          <div>
+            {toolbarText}
+            <div>
+              <Suspense fallback="Loading dynamic toolbar content...">
+                <DynamicText text="Toolbar content" />
+              </Suspense>
+            </div>
+          </div>
+        );
+      }
+
+      function App() {
+        // Note that Toolbar and MainContent are in two completely separate
+        // Suspense trees. One should not block the other.
+        return (
+          <div>
+            <Suspense fallback="Loading static toolbar shell...">
```

Comment:
I think arguably this outer Suspense boundary shouldn't be necessary to unblock "Main content" during the dynamic render, but I kept it here to make it super clear that we're dealing with completely parallel trees.